packages/next/build/webpack-config.ts

```diff
@@ -373,6 +370,7 @@ export default function getBaseWebpackConfig (dir: string, {dev = false, debug =
 
       return entry
     }
+    if (!dev) webpackConfig.entry = await webpackConfig.entry()
```

Comment:
Why is this inside `if (typeof originalEntry !== 'undefined') {` 🤔

Reply:
It should be above the `return webpackConfig`

Reply:
@timneutkens Probably because of the typescript type set for `webpack.entry` that allows `undefined`, it's set as `any` so not required anyway, will remove it